src/main/drivers/stm32/timer_stm32h5xx.c

```diff
@@ -0,0 +1,225 @@
+    DEF_TIM(TIM4, CH2, PD13, 0, 0, 0),
+    DEF_TIM(TIM4, CH3, PD14, 0, 0, 0),
+    DEF_TIM(TIM4, CH4, PD15, 0, 0, 0),
+
+// Port E
+    DEF_TIM(TIM1, CH1N, PE8, 0, 0, 0), // AF1
+    DEF_TIM(TIM1, CH1, PE9, 0, 0, 0),
+    DEF_TIM(TIM1, CH2N, PE10, 0, 0, 0),
+    DEF_TIM(TIM1, CH2, PE11, 0, 0, 0),
+    DEF_TIM(TIM1, CH3N, PE12, 0, 0, 0),
+    DEF_TIM(TIM1, CH3, PE13, 0, 0, 0),
+    DEF_TIM(TIM1, CH4, PE14, 0, 0, 0),
+
+    DEF_TIM(TIM1, CH4N, PE15, 0, 0, 0), // AF3 (ADDED)
+
+    DEF_TIM(TIM15, CH1N, PE4, 0, 0, 0), // AF4
+    DEF_TIM(TIM15, CH1, PE5, 0, 0, 0),
+    DEF_TIM(TIM15, CH2, PE6, 0, 0, 0),
+
+// Port F
+    DEF_TIM(TIM16, CH1, PF6, 0, 0, 0), // AF1
+    DEF_TIM(TIM17, CH1, PF7, 0, 0, 0),
+    DEF_TIM(TIM16, CH1N, PF8, 0, 0, 0), // AF3
+    DEF_TIM(TIM17, CH1N, PF9, 0, 0, 0),
+
+    DEF_TIM(TIM13, CH1, PF8, 0, 0, 0), // AF9 (WAS TIM13_CH1N)
+    DEF_TIM(TIM14, CH1, PF9, 0, 0, 0), // AF9 (WAS TIM14_CH1N)
+
+// Port H
+
+    DEF_TIM(TIM1, CHN3, PH6, 0, 0, 0), // AF1 (ADDED)
+    DEF_TIM(TIM1, CH3, PH7, 0, 0, 0),
+    DEF_TIM(TIM1, CHN2, PH8, 0, 0, 0),
+    DEF_TIM(TIM1, CH2, PH9, 0, 0, 0),
+    DEF_TIM(TIM1, CHN1, PH10, 0, 0, 0),
+    DEF_TIM(TIM1, CH1, PH11, 0, 0, 0),
+
+    DEF_TIM(TIM12, CH1, PH6, 0, 0, 0), // AF2 (ADDED)
+    DEF_TIM(TIM12, CH2, PH9, 0, 0, 0),
+    DEF_TIM(TIM5, CH1, PH10, 0, 0, 0),
+    DEF_TIM(TIM5, CH2, PH11, 0, 0, 0),
+    DEF_TIM(TIM5, CH3, PH12, 0, 0, 0),
+
+    DEF_TIM(TIM8, CH1, PH6, 0, 0, 0), // AF3 (ADDED)
+    DEF_TIM(TIM8, CH1N, PH7, 0, 0, 0),
+    DEF_TIM(TIM8, CH2, PH8, 0, 0, 0),
+    DEF_TIM(TIM8, CH2N, PH9, 0, 0, 0),
+    DEF_TIM(TIM8, CH3, PH10, 0, 0, 0),
+    DEF_TIM(TIM8, CH3N, PH11, 0, 0, 0),
+    DEF_TIM(TIM8, CH1N, PH13, 0, 0, 0),
+    DEF_TIM(TIM8, CH2N, PH14, 0, 0, 0),
+    DEF_TIM(TIM8, CH3N, PH15, 0, 0, 0),
+
+// Port I
+    DEF_TIM(TIM5, CH4, PI0, 0, 0, 0), // AF2 (ADDED)
+
+    DEF_TIM(TIM8, CH4, PI2, 0, 0, 0), // AF3 (ADDED)
+    DEF_TIM(TIM8, CH1, PI5, 0, 0, 0),
+    DEF_TIM(TIM8, CH2, PI6, 0, 0, 0),
+    DEF_TIM(TIM8, CH3, PI7, 0, 0, 0),
+};
+#endif
+
+
+uint32_t timerClock(const TIM_TypeDef *tim)
+{
+    int timpre;
+    uint32_t pclk;
+    uint32_t ppre;
+
+    // This function is used to calculate the timer clock frequency.
+    // RM0481 (Rev 3) Table 
+
+#define PERIPH_PRESCALER(bus) ((RCC->D2CFGR & RCC_D2CFGR_D2PPRE ## bus) >> RCC_D2CFGR_D2PPRE ## bus ## _Pos)
+
+    if (tim == TIM1 || tim == TIM8 || tim == TIM15 || tim == TIM16 || tim == TIM17) {
+        // Timers on APB2
+        pclk = HAL_RCC_GetPCLK2Freq();
+        ppre = PERIPH_PRESCALER(2);
+    } else {
+        // Timers on APB1
+        pclk = HAL_RCC_GetPCLK1Freq();
+        ppre = PERIPH_PRESCALER(1);
+    }
+
+    timpre = (RCC->CFGR & RCC_CFGR_TIMPRE) ? 1 : 0;
+
+    int index = (timpre << 3) | ppre;
+
+    static uint8_t periphToKernel[2][8] = { // The multiplier table
+        { 1, 1, 1, 1, 2, 2, 2, 2 }, // TIMPRE = 0
+        { 1, 1, 1, 1, 2, 4, 4, 4 }  // TIMPRE = 1
```

Comment:
```suggestion
    RCC_ClkInitTypeDef  clkConfig, uint32_t fLatency
    HAL_RCC_GetClockConfig(&clkConfig, &fLatency);
    if ((uintptr_t)tim >= APB2PERIPH_BASE ) {  // APB2 
        pclk = HAL_RCC_GetPCLK2Freq();
        ppre = clkConfig.APB2CLKDivider;
    } else {  // all other timers are on APB1
        pclk = HAL_RCC_GetPCLK1Freq();
        ppre = clkConfig.APB1CLKDivider;
    }

    timpre = (RCC->CFGR & RCC_CFGR_TIMPRE) ? 1 : 0;

#define PC(m) (0x80 | (m))  .. multiply pclk
#define HC(m) (0x00 | (m))  .. multiply hclk
        static const uint8_t periphToKernel[2][8] = { // The multiplier table
      //  1      1      1      1      2      4      8      16 
        { HC(1), HC(1), HC(1), HC(1), HC(1), PC(2), PC(2), PC(2) }, // TIMPRE = 0 
        { PC(2), PC(2), PC(2), PC(2), PC(2), PC(2), PC(4), PC(4) }     // TIMPRE = 1 
      }
#undef PC
#undef HC
     int flagMult = periphToKernel[timpre][ppre];
     if (flagMult & 0x80) { // PCLK based
         return pclk * (flagMult & 0x7f);
     } else {
         return HAL_RCC_GetHCLKFreq() *  (flagMult & 0x7f);
     }
         
```

I think that periphToKernel should look this way on H5

HAL_RCC_GetClockConfig will remove a magical macro and can be used for other STM families, but periphToKernel is family-dependent. 

Original table is wrong for H7 (RM0410: 1:If the APB prescaler (PPRE1, PPRE2 in the RCC_CFGR register) is configured to a
division factor of 1, 2 or 4, TIMxCLK = HCLK. Otherwise, the timer clock frequencies are set
to four times to the frequency of the APB domain to which the timers are connected:
TIMxCLK = 4xPCLKx.)

_PPRE2[2:0]: APB high-speed prescaler (APB2)
Set and reset by software to control APB high-speed clocks division factor.
The clocks are divided with the new prescaler factor from 1 to 16 APB cycles after PPRE2
write.
0xx: rcc_pclk2 = rcc_hclk1
100: rcc_pclk2 = rcc_hclk1 / 2
101: rcc_pclk2 = rcc_hclk1 / 4
110: rcc_pclk2 = rcc_hclk1 / 8
111: rcc_pclk2 = rcc_hclk1 / 16

TIMPRE: timers clocks prescaler selection
This bit is set and reset by software to control the clock frequency of all the timers connected
to APB1 and APB2 domains.
0: The timers kernel clock is equal to rcc_hclk1 if PPRE1 or PPRE2 corresponds to a
division by 1 or 2, else it is equal to 2 x Frcc_pclk1 or 2 x Frcc_pclk2 (default after reset)
1: The timers kernel clock is equal to 2 x Frcc_pclk1 or 2 x Frcc_pclk2 if PPRE1 or PPRE2
corresponds to a division by 1, 2 or 4, else it is equal to 4 x Frcc_pclk1 or 4 x Frcc_pclk2_